### PR TITLE
assorted little cleanups because I have a soft spot for compiler warnings

### PR DESCRIPTION
**Pull request by d**
See messages in each commit.

## Here are some reminders before you submit the pull request
- [x] Add tests for the change
- [x] Communicate in the mailing list if needed
- [x] Pass `make installcheck`

### CODE REVIEW

**Comment by Eulerizeit** [comment]
What is the definition of done for this?

Are you committing this to 5 because these are fixed on Master?

**Comment by d** [comment]
> re you committing this to 5 because these are fixed on Master?

Yes, these are mostly compiler warnings (and I'm sad to report that the compiler is correct more often than not) that only show up on 5X_STABLE

**Comment by d** [comment]
Reviewers, I failed to convey that I'm not done yet

**Comment by d** [comment]
It's past Tuesday, and we still have a ton of warnings mostly around the usage of `printf` (format overflow) and `sprintf` (`restrict` violation). But I'm gonna stop working on this. @asimrp and @hlinnaka , wanna give those a second look?

**Comment by hlinnaka** [comment]
LGTM, except for the conflict marker in c.h

File: src/include/c.h [hunk 1]
```diff
@@ -939,7 +939,14 @@ typedef NameData *Name;
  */
 
 #if !HAVE_DECL_SNPRINTF
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (d) [reply src/include/c.h]: OMFG this is embarrassing. Fixed it.

**Comment by d** [comment]
Pushed. Thanks everyone for the review. Removing that conflict marker is good (IDE does the correct magic the second time around), phew...

**Pull request merged**